pkg/apis/machine/validation/awsmachineclass.go

```diff
@@ -132,12 +132,19 @@ func validateBlockDevices(blockDevices []machine.AWSBlockDeviceMappingSpec, fldP
 
 	validVolumeTypes := []string{"gp2", "io1", "st1", "sc1", "standard"}
 
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
+	const dataDeviceNameFmt string = `^/dev/(sd[a-z]|xvd[a-c][a-z]?)$`
+	var dataDeviceNameRegexp = regexp.MustCompile("^" + dataDeviceNameFmt + "$")
```

Comment:
Do `^` and `$` still need to be there if `dataDeviceNameFmt` already has them?

Reply:
fixed